Server/mods/deathmatch/logic/CPlayer.cpp

```diff
@@ -512,11 +512,39 @@ unsigned char CPlayer::GetAttackBodyPart()
     return m_ucAttackBodyPart;
 }
 
-void CPlayer::SetTeam(CTeam* pTeam, bool bChangeTeam)
+void CPlayer::SetTeam(CTeam* pTeam, bool bChangeTeam, CTeam::eChangeReason Reason)
 {
     if (pTeam == m_pTeam)
         return;
 
+    // If the new team is NULL, this means he was removed from his previous team
+    if (Reason == CTeam::CHANGE_CHANGED && pTeam == NULL)
+        Reason = CTeam::CHANGE_REMOVED;
+
+    // Grab change reason
+    const char* szReason = "unknown";
+    switch (Reason)
+    {
+        case CTeam::CHANGE_CHANGED:
+            szReason = "changed";
+            break;
+        case CTeam::CHANGE_REMOVED:
+            szReason = "removed";
+            break;
+        case CTeam::CHANGE_DISCONNECTED:
+            szReason = "disconnected";
+            break;
+        case CTeam::CHANGE_DESTROYED:
+            szReason = "destroyed";
+            break;
+    }
```

Comment:
I'd implement `EnumToString` for `CTeam::ChangeReason` instead, and use that, as its more universal.
Altough, im not sure where you should add the `IMPLEMENT_ENUM` things for it, I'd say just stash it into CTeams.cpp (and h) somewhere. (See CLuaFunctionParseHelpers.cpp for some example code)